mbodied/robots/robot.py

```diff
@@ -15,21 +15,40 @@
 import asyncio
 from abc import ABC, abstractmethod
 
+from mbodied.robots.robot_recording import RobotRecorder
 from mbodied.types.sample import Sample
 
 
 class Robot(ABC):
     """Abstract base class for robot hardware interfaces.
 
-    This class provides a template for creating robot hardware interfaces that can
-    control robots or other hardware devices.
-
-    Methods:
-        do (required): Executes motion on the robot.
-        fetch: Fetches data from the hardware.
-        capture: Captures continuous data from the hardware i.e. image.
-        get_robot_state: Gets the current pose of the hardware, optional for robot recorder.
-        calculate_action: Calculates the action between two robot states, optional for robot recorder.
+    This class serves as a blueprint for creating interfaces to control robot hardware
+    or other devices. It provides essential methods and guidelines to implement
+    functionalities such as executing motions, capturing observations, and recording
+    robot data, which can be used for training models.
+
+    Key Features:
+    - **Motion Execution:** Implement the `do` method to execute robot movements.
+    - **Data Fetching:** Implement `fetch` to retrieve data from the hardware.
+    - **Continuous Data Capture:** Implement `capture` for continuous data capture.
```

Comment:
Key feature:
Its not a blueprint or essential methods.

-organize your hardware interface in a modular fashion
-support asynchronous dataset creation


Why not just name methods observer(), get_state(), prepare_action() (or post-process motion()) and do() ?

By the way you shouldn't need to calculate_action() thats the whole point of defining a Motion.

Reply:
i renamed them to get_observation(), get_state() and prepare_action()